tasks/backfill_existing_gh_app_installations.py

```diff
@@ -0,0 +1,122 @@
+import logging
+from typing import List, Optional
+
+from sqlalchemy.orm.session import Session
+
+from app import celery_app
+from celery_config import backfill_existing_gh_app_installations_name
+from database.models.core import GithubAppInstallation, Owner
+from helpers.backfills import (
+    add_repos_service_ids_from_provider,
+    maybe_set_installation_to_all_repos,
+)
+from services.owner import get_owner_provider_service
+from tasks.base import BaseCodecovTask
+
+log = logging.getLogger(__name__)
+
+yield_amount = 100
+
+
+class BackfillExistingGHAppInstallationsTask(
+    BaseCodecovTask, name=backfill_existing_gh_app_installations_name
+):
+    def backfill_existing_gh_apps(
+        self, db_session: Session, owner_ids: List[int] = None, missed_owner_ids=[]
+    ):
+        # Get owners that have installations, and installations queries
+        owners_query = (
+            db_session.query(Owner)
+            .join(GithubAppInstallation, Owner.ownerid == GithubAppInstallation.ownerid)
+            .filter(
+                Owner.service == "github",
+            )
+        )
+        gh_app_installations_query = db_session.query(GithubAppInstallation)
+
+        # Filter if owner_ids were provided
+        if owner_ids:
+            owners_query = owners_query.filter(Owner.ownerid.in_(owner_ids))
+            gh_app_installations_query = gh_app_installations_query.filter(
+                GithubAppInstallation.ownerid.in_(owner_ids)
+            )
+
+        owners: List[Owner] = owners_query.all()
+        gh_app_installations: List[
+            GithubAppInstallation
+        ] = gh_app_installations_query.yield_per(yield_amount)
+
+        # I need to make reference of these in the gh installation app, Ill take suggestions here
+        owners_dict = {owner.ownerid: owner for owner in owners}
+
+        for gh_app_installation in gh_app_installations:
+            # Check if gh app has 'all' repositories selected
+            owner = owners_dict[gh_app_installation.ownerid]
```

Comment:
Shouldn't you be able to get the owner directly from the `GithubAppInstallation` instance?

https://github.com/codecov/worker/blob/bcec3478e2a27bb7950f40388cf191834fb2d5a3/database/models/core.py#L193

Reply:
Good catch, will get rid of this